### PR TITLE
Add xorg protocols

### PR DESCRIPTION
**Pull request by flyingP0tat0**
One step towards #889.

### CODE REVIEW

**Comment by uberhacker** [comment]
Wow, that is a ton of new packages.  Nice start.  I assume you are planning on adding description and homepage later?

**Comment by flyingP0tat0** [comment]
@uberhacker Done.

File: packages/xcb_proto.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require 'package'
+
+class Xcb_proto < Package
+  description 'The protocols for the X window system provide extended functionality for communication between a X client and the server.'
+  homepage 'https://x.org'
+  version '1.12'
+  source_url 'https://www.x.org/archive/individual/xcb/xcb-proto-1.12.tar.gz'
+  source_sha256 'cfa49e65dd390233d560ce4476575e4b76e505a0e0bacdfb5ba6f8d0af53fd59'
+
+  depends_on 'python'
```
Comment (yancouto) [inline packages/xcb_proto.rb]: Shouldn't this be changed to python27?
When running the ./configure, it found python with version 2.7
```
checking for python... /usr/local/bin/python
checking for python version... 2.7
```
(and the 'python' package installs python3, so this won't influence it, or maybe it accepts both)

I'm not sure, python is confusing (#923 is relevant)

Comment (flyingP0tat0) [inline packages/xcb_proto.rb]: I tested it with python(3) and it works fine.

Comment (jam7) [inline packages/xcb_proto.rb]: since you are using x86_64.  it doesnt on arm.  please use either python3 or python27 as one of dependencies.

**Comment by jam7** [comment]
When I try it with python3, it shows following erros.  python27 works without problem, though.
```
$ crew install xcb_proto
...
checking for a Python interpreter with version >= 2.5... python3                                                                      
checking for python3... /usr/local/bin/python3                              
checking for python3 version... 3.6                                                                                                   
checking for python3 platform... linux                                      
checking for python3 script directory... ${prefix}/lib/python3.6/site-packages                                                        
checking for python3 extension module directory... ${exec_prefix}/lib/python3.6/site-packages
...
make[2]: Nothing to be done for 'install-exec-am'.                          
 /bin/mkdir -p '/usr/local/tmp/crew/dest/usr/local/lib/python3.6/site-packages/xcbgen'                                                
 /usr/bin/install -c -m 644 __init__.py error.py expr.py align.py matcher.py state.py xtypes.py '/usr/local/tmp/crew/dest/usr/local/lib/python3.6/site-packages/xcbgen'                                                                                                     
Byte-compiling python modules...                                            
__init__.pyerror.pyexpr.pyalign.pySorry: TabError: inconsistent use of tabs and spaces in indentation (align.py, line 19)             
matcher.pystate.pyxtypes.py  File "/usr/local/lib/python3.6/site-packages/xcbgen/xtypes.py", line 504
    print "Explicit start-align for %s: %s\n" % (self, self.required_start_align)                                                     
                                            ^                               
SyntaxError: invalid syntax                                                                                                           
                                                                            
                                                                                                                                      
Byte-compiling python modules (optimized versions) ...                      
__init__.pyerror.pyexpr.pyalign.pymatcher.pystate.pyxtypes.py                                                                         
make[2]: Leaving directory '/usr/local/tmp/crew/xcb-proto-1.12.tar.gz.dir/xcb-proto-1.12/xcbgen'
make[1]: Leaving directory '/usr/local/tmp/crew/xcb-proto-1.12.tar.gz.dir/xcb-proto-1.12/xcbgen'
```

**Comment by flyingP0tat0** [comment]
@jam7 Ok, changed.

**Pull request merged**